### PR TITLE
Replace CLA by DCO in review guidelines

### PR DESCRIPTION
**Pull request by Miouge1**
**What this PR does / why we need it**:
CLA is replaced by DCO, see https://www.helm.sh/blog/helm-dco/index.html

**Which issue this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*: fixes #

**Special notes for your reviewer**:

### CODE REVIEW

**Comment by cpanato** [comment]
i have no objections on that

/lgtm

File: REVIEW_GUIDELINES.md [hunk 1]
```diff
@@ -4,7 +4,7 @@ Anyone is welcome to review pull requests. Besides our [technical requirements](
 
 ## Process
 
-The process to get a pull request merged is fairly simple. First, all required tests need to pass and the contributor needs to have a signed CLA. See [Charts Testing](https://github.com/helm/charts/blob/master/test/README.md) for details on our CI system and how you can provide custom values for testing. If there is a problem with some part of the test, such as a timeout issue, please contact one of the charts repository maintainers by commenting `cc @helm/charts-maintainers`.
+The process to get a pull request merged is fairly simple. First, all required tests need to pass and the contributor needs to have a signed DCO. See [Charts Testing](https://github.com/helm/charts/blob/master/test/README.md) for details on our CI system and how you can provide custom values for testing. If there is a problem with some part of the test, such as a timeout issue, please contact one of the charts repository maintainers by commenting `cc @helm/charts-maintainers`.
```
Comment (davidkarlsen) [inline REVIEW_GUIDELINES.md]: would it be good to add a link to DCO: https://www.helm.sh/blog/helm-dco/index.html to make more information available w/o search?

Comment (Miouge1) [reply REVIEW_GUIDELINES.md]: Good idea, i'm adding in the latest commit.

**Pull request merged**